docker/docker_client.go

```diff
@@ -214,7 +214,9 @@ func dockerCertDir(sys *types.SystemContext, hostPort string) (string, error) {
 // signatureBase is always set in the return value
 func newDockerClientFromRef(sys *types.SystemContext, ref dockerReference, write bool, actions string) (*dockerClient, error) {
 	registry := reference.Domain(ref.ref)
-	auth, err := config.GetCredentials(sys, registry)
+	refPath := reference.Path(ref.ref)
+
+	auth, err := config.GetCredentialsWithRefPath(sys, registry, refPath)
```

Comment:
I think it would be more idiomatic to pass a `reference.Reference` around than splitting/joining.

Reply:
I've chosen a `reference.Named`, which I hope is fine as well since it's a more concrete type.

Reply:
Yes, even better!